pkg/ruler/configs.go

```diff
@@ -0,0 +1,105 @@
+package ruler
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/weaveworks/cortex/pkg/configs"
+	configs_client "github.com/weaveworks/cortex/pkg/configs/client"
+	"github.com/weaveworks/cortex/pkg/configs/db"
+	"github.com/weaveworks/cortex/pkg/util"
+)
+
+// ConfigStoreConfig says where we can find the ruler configs.
+type ConfigStoreConfig struct {
+	DBConfig db.Config
+
+	// DEPRECATED
+	ConfigsAPIURL util.URLValue
+
+	// DEPRECATED. HTTP timeout duration for requests made to the Weave Cloud
+	// configs service.
+	ClientTimeout time.Duration
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet
+func (cfg *ConfigStoreConfig) RegisterFlags(f *flag.FlagSet) {
+	cfg.DBConfig.RegisterFlags(f)
+	f.Var(&cfg.ConfigsAPIURL, "ruler.configs.url", "DEPRECATED. URL of configs API server.")
+	f.DurationVar(&cfg.ClientTimeout, "ruler.client-timeout", 5*time.Second, "DEPRECATED. Timeout for requests to Weave Cloud configs service.")
+}
+
+// RulesAPI is what the ruler needs from a config store to process rules.
+type RulesAPI interface {
+	// GetConfigs returns all Cortex configurations from a configs API server
+	// // TODO: hat have been updated after the given configs.ID was last updated.
+	GetConfigs(since configs.ID) (map[string]configs.VersionedRulesConfig, error)
+}
+
+// NewRulesAPI creates a new RulesAPI.
+func NewRulesAPI(cfg ConfigStoreConfig) (RulesAPI, error) {
+	// All of this falderal is to allow for a smooth transition away from
+	// using the configs server and toward directly connecting to the database.
+	// See https://github.com/weaveworks/cortex/issues/619
+	if cfg.DBConfig.URI == "" {
+		return configsClient{
+			URL:     cfg.ConfigsAPIURL.URL,
+			Timeout: cfg.ClientTimeout,
+		}, nil
+	}
+	db, err := db.NewRulesDB(cfg.DBConfig)
+	if err != nil {
+		return nil, err
+	}
+	return dbStore{db: db}, nil
+}
+
+// configsClient allows retrieving recording and alerting rules from the configs server.
+type configsClient struct {
+	URL     *url.URL
+	Timeout time.Duration
+}
+
+// GetConfigs implements RulesAPI.
+func (c configsClient) GetConfigs(since configs.ID) (map[string]configs.VersionedRulesConfig, error) {
+	suffix := ""
+	if since != 0 {
```

Comment:
It's for rules, not alertmanager configs, but "why doesn't this use `RulerConfigsAPI.GetConfigs()`?" is a valid question. The answer is that it replaces `RulerConfigsAPI`, and has a slightly different API (`configs.VersionedRulesConfig` is new in this PR).

We're replacing it because we don't want to have a client library per se. The ruler will be directly responsible for querying the database, so there's no need for code in another package that knows how to get ruler configs.